addons/account/wizard/account_resequence.py

```diff
@@ -0,0 +1,160 @@
+                if record.sequence_number_reset == 'year':
+                    sequence = re.match(self.env['account.move']._sequence_yearly_regex, record.first_name)
+                    format = '{prefix1}%(year)04d{prefix2}%(seq)0{len}d{suffix}'.format(
+                        prefix1=sequence.group('prefix1'),
+                        prefix2=sequence.group('prefix2'),
+                        len=len(sequence.group('seq')),
+                        suffix=sequence.group('suffix'),
+                    )
+                elif record.sequence_number_reset == 'month':
+                    sequence = re.match(self.env['account.move']._sequence_monthly_regex, record.first_name)
+                    format = '{prefix1}%(year)04d{prefix2}%(month)02d{prefix3}%(seq)0{len}d{suffix}'.format(
+                        prefix1=sequence.group('prefix1'),
+                        prefix2=sequence.group('prefix2'),
+                        prefix3=sequence.group('prefix3'),
+                        len=len(sequence.group('seq')),
+                        suffix=sequence.group('suffix'),
+                    )
+                else:
+                    sequence = re.match(self.env['account.move']._sequence_fixed_regex, record.first_name)
+                    format = '{prefix}%(seq)0{len}d{suffix}'.format(
+                        prefix=sequence.group('prefix1'),
+                        len=len(sequence.group('seq')),
+                        suffix=sequence.group('suffix'),
+                    )
+            except AttributeError:
+                continue
+
+            new_values = {}
+            for j, grouped_list in enumerate(sorted.values()):
+                for move in grouped_list:
+                    new_values[move.id] = {
+                        'current_name': move.name,
+                        'state': move.state,
+                        'date': format_date(self.env, move.date),
+                        'server-date': str(move.date),
+                    }
+
+                new_name_list = [format % {
+                    'year': grouped_list[0].date.year,
+                    'month': grouped_list[0].date.month,
+                    'seq': i + (int(sequence.group('seq') or '1') if j == (len(sorted)-1) else 1),
+                } for i in range(len(grouped_list))]
+
+                for move, new_name in zip(grouped_list.sorted(lambda m: _sort_by_name_key(m.name)), new_name_list):
+                    new_values[move.id]['new_by_name'] = new_name
+                for move, new_name in zip(grouped_list.sorted(lambda m: (m.date, m.name, m.id)), new_name_list):
+                    new_values[move.id]['new_by_date'] = new_name
+
+            record.new_values = json.dumps(new_values)
+
+    def resequence(self):
+        new_values = json.loads(self.new_values)
+        self.with_context(tracking_disable=True).move_ids.state = 'draft'
+        for move_id in self.move_ids:
+            if str(move_id.id) in new_values:
```

Comment:
str(move_id.id) ?? why not move_id.id directly?

Reply:
because of the `json.loads`. Integers are converted to strings because json keys are always strings